staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -201,7 +201,19 @@ func (le *LeaderElector) renew() {
 	stop := make(chan struct{})
 	wait.Until(func() {
 		err := wait.Poll(le.config.RetryPeriod, le.config.RenewDeadline, func() (bool, error) {
```

Comment:
Poll already retries and timesout. What does this do differently?

Reply:
Indeed wait.Poll check timeout before running `ConditionFunc`. If ConditionFunc takes too long, it can not quit quickly.